.github/workflows/main.yml

```
@@ -36,46 +36,56 @@ jobs:
           cd apps
           bash build_all.sh
 
-      - name: Test workflow dataset-ingestion
+  test-apps:
+    needs:
+      - build-apps
+
+    strategy:
+      matrix:
+        app: [
+          dataset-ingestion,
+          object-detection,
+          embeddings-extraction,
+          face-detection,
+          jupyterlab,
+          rag
+        ]
+    runs-on:
+      - deployer
+      - self-hosted
```

```suggestion
      - ubuntu-latest
```
For this particular case, it may be better to use ubuntu-latest, provided by github, because it has infinite (not really, but you get the idea) parallelism. This is compared to our internal runners, which can run fewer jobs in parallel (4 or 5, but assuming nothing else is running). The tradeoff is that our runners have more horse power. 
For this job only (`test-apps`), I think it makes sense to use ubuntu-latest runners.

---

yes, sounds good let's try